test/test_autosplit.py

```diff
@@ -23,19 +23,21 @@
 class ExampleCode(torch.nn.Module):
```

Comment:
can you add another test case inspired by torchtrain, where module has a ModuleList like this

Transformer
        self.layers = torch.nn.ModuleList()
        for layer_id in range(model_args.n_layers):
            self.layers.append(TransformerBlock(layer_id, model_args))

What I am wondering is, how will you deal with  keeping the FQNs the same after splitting?

if you put layers[0] on one stage and layers[1] on next stage,

will second stage still have layers[1] as FQN or will it drop to layers[0]?

Reply:
Thanks. Good idea. 
The second stage will have "layers.1" as the FQN.
"layers" is from the `self.layers = torch.nn.ModuleList()` tier.
"1" corresponds to the attribute within the ModuleList.
Those two are preserved from the original model's FQN.

Reply:
@wconstab I added `test_transformer.py`. 
You can see the updated PR description for the split structure.